src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -457,80 +421,43 @@ private IEnumerable<(ManifestId id, ManifestVersion version, SdkFeatureBand feat
+            var versions = WorkloadSet.FromJson(fileContent, _sdkFeatureBand).ManifestVersions;
+            return versions.Select(kvp => (kvp.Key, new ManifestVersionWithBand(kvp.Value.Version, kvp.Value.FeatureBand)));
         }
 
-        private bool BackgroundUpdatesAreDisabled() =>
-            bool.TryParse(_getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_UPDATE_NOTIFY_DISABLE), out var disableEnvVar) && disableEnvVar;
+        private bool BackgroundUpdatesAreDisabled() => bool.TryParse(_getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_UPDATE_NOTIFY_DISABLE), out var disableEnvVar) && disableEnvVar;
 
         private string GetAdvertisingManifestSentinelPath(SdkFeatureBand featureBand) => Path.Combine(_userProfileDir, $".workloadAdvertisingManifestSentinel{featureBand}");
 
         private string GetAdvertisingWorkloadsFilePath(SdkFeatureBand featureBand) => GetAdvertisingWorkloadsFilePath(_userProfileDir, featureBand);
 
         private static string GetAdvertisingWorkloadsFilePath(string userProfileDir, SdkFeatureBand featureBand) => Path.Combine(userProfileDir, $".workloadAdvertisingUpdates{featureBand}");
 
-        private async Task<string> GetOnlinePackagePath(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews)
-        {
-            string packagePath = await _nugetPackageDownloader.DownloadPackageAsync(
-                _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand),
-                packageSourceLocation: _packageSourceLocation,
-                includePreview: includePreviews);
-
-            return packagePath;
-        }
-
-        private string GetOfflinePackagePath(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, DirectoryPath? offlineCache = null)
-        {
-            string packagePath = Directory.GetFiles(offlineCache.Value.Value)
-                .Where(path => path.EndsWith(".nupkg"))
-                .Where(path =>
-                {
-                    var manifestPackageId = _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand).ToString();
-                    return Path.GetFileName(path).StartsWith(manifestPackageId, StringComparison.OrdinalIgnoreCase);
-                })
-                .Max();
-
-            return packagePath;
-        }
-
-        private async Task<(bool, string)> GetManifestPackageUpdate(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews, DirectoryPath? offlineCache = null)
+        private async Task<(bool Success, string PackagePath)> GetManifestPackageUpdate(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews, DirectoryPath? offlineCache = null)
         {
-            if (offlineCache == null || !offlineCache.HasValue)
+            var manifestPackageId = _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand);
+            if (offlineCache != null)
```

Comment:
Do you still need the HasValue check?

Reply:
For something using `Nullable<T>`, checking if it is `null` or checking if it has a value is the same thing. The "nullness" of the class represents if it has a value of T. https://stackoverflow.com/a/676089/294804